test/sjcl-ecdsa-signdeterministic-test.js

```diff
@@ -0,0 +1,184 @@
+      k: '103f90ee9dc52e5e7fb5132b7033c63066d194321491862059967c715985d473',
+      r: '53b2fff5d1752b2c689df257c04c40a587fababb3f6fc2702f1343af7ca9aa3f',
+      s: 'b9afb64fdc03dc1a131c7d2386d11e349f070aa432a4acc918bea988bf75c74c'
+    },*/ {
+      hash: sjcl.hash.sha256,
+      message: 'sample',
+      k: 'a6e3c57dd01abe90086538398355dd4c3b17aa873382b0f24d6129493d8aad60',
+      r: 'efd48b2aacb6a8fd1140dd9cd45e81d69d2c877b56aaf991c34d0ea84eaf3716',
+      s: 'f7cb1c942d657c41d436c7a1b6e29f65f3e900dbb9aff4064dc4ab2f843acda8'
+    },/* {
+      hash: sjcl.hash.sha384,
+      message: 'sample',
+      k: '09f634b188cefd98e7ec88b1aa9852d734d0bc272f7d2a47decc6ebeb375aad4',
+      r: '0eafea039b20e9b42309fb1d89e213057cbf973dc0cfc8f129edddc800ef7719',
+      s: '4861f0491e6998b9455193e34e7b0d284ddd7149a74b95b9261f13abde940954'
+    },*/ {
+      hash: sjcl.hash.sha512,
+      message: 'sample',
+      k: '5fa81c63109badb88c1f367b47da606da28cad69aa22c4fe6ad7df73a7173aa5',
+      r: '8496a60b5e9b47c825488827e0495b0e3fa109ec4568fd3f8d1097678eb97f00',
+      s: '2362ab1adbe2b8adf9cb9edab740ea6049c028114f2460f96554f61fae3302fe'
+    }
+  ];
+
+  function testRFC(test, index) {
+    it('Should create the right k and s for RFC test #' + index, function() {
+
+      var hashed_message = test.hash.hash(test.message);
+      //var k = rfcPrivate.generateK(hashed_message, test.hash, rfcQ);
+      var k = rfcPrivate.deterministicGenerateK(rfcPrivate._curve, hashed_message, rfcPrivate, test.hash);
+
+      console.log('generated k: ' + sjcl.codec.hex.fromBits(k.toBits()));
+      console.log('test k:      ' + test.k);
+
+      var signature = rfcPrivate.signDeterministic(hashed_message, test.hash);
+      var r = sjcl.bitArray.bitSlice(signature, 0, rfcQlen);
+      var s = sjcl.bitArray.bitSlice(signature, rfcQlen);
+    
+      console.log('generated r: ' + sjcl.codec.hex.fromBits(r));
+      console.log('test r:      ' + test.r);
+
+      console.log('generated s: ' + sjcl.codec.hex.fromBits(s));
+      console.log('test s:      ' + test.s);
+
+      assert.equal(sjcl.codec.hex.fromBits(k.toBits()), test.k);
+
+      assert.equal(sjcl.codec.hex.fromBits(s), test.s);
+      assert.equal(sjcl.codec.hex.fromBits(r), test.r);
+      /* make sure that the signature is correct too */
+      assert(rfcPublic.verify(hashed_message, signature));
+    });
+  }
+  rfcTests.forEach(testRFC);
+
+  it('Should create the same signatures for the same inputs', function() {
+
+    var message = 'Hello world!';
+    var secret_hex =
+      '9931c08f61f127d5735fa3c60e702212ce7ed9a2ac90d5dbade99c689728cd9b';
+    var hash_function = sjcl.hash.sha256;
+    var k256_curve = sjcl.ecc.curves.k256;
+
+    var old_signature;
+
+    for (var i = 0; i < 4; i++) {
+
+      var secret_bn = sjcl.bn.fromBits(sjcl.codec.hex.toBits(secret_hex));
+      var secret_key = new sjcl.ecc.ecdsa.secretKey(k256_curve, secret_bn);
+      var hashed_message = hash_function.hash(message);
+
```

Comment:
I'd personally find it easier to read if you just move this block:
```
+      var secret_bn = sjcl.bn.fromBits(sjcl.codec.hex.toBits(secret_hex));
+      var secret_key = new sjcl.ecc.ecdsa.secretKey(k256_curve, secret_bn);
+      var hashed_message = hash_function.hash(message);
```
outside of the loop and just did:
```
var sig1 = secret_key.signDeterministic(hashed_message,  hash_function);
var sig2 = secret_key.signDeterministic(hashed_message,  hash_function);
assert.equal(sig1, sig2);
```

instead of a loop.